projects/common/element-components/spell-correct.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, ViewChild } from '@angular/core';
+import { MatInput } from '@angular/material/input';
+import { FormElementComponent } from '../form-element-component.directive';
+import { SpellCorrectElement } from '../models/spell-correct-element';
+
+@Component({
+  selector: 'app-spell-correct',
+  template: `
+    <div fxFlex
+         fxLayout="column"
+         [style.width.%]="100"
+         appInputBackgroundColor [backgroundColor]="elementModel.backgroundColor"
+         [style.height.%]="100">
+      <mat-form-field class="small-input">
+        <input matInput type="text"
+               [style.text-align]="'center'"
+               autocomplete="off"
+               [formControl]="elementFormControl">
+      </mat-form-field>
+      <button mat-button
+              [style.color]="elementModel.fontColor"
+              [style.font-family]="elementModel.font"
+              [style.font-size.px]="elementModel.fontSize"
+              [style.font-weight]="elementModel.bold ? 'bold' : ''"
+              [style.font-style]="elementModel.italic ? 'italic' : ''"
+              [style.width.%]="100"
+              [style.margin-top]="'-20px'"
+              [style.text-decoration-line]="strikethrough() ? 'line-through' : ''"
+              (click)="onClick($event)">
+        &nbsp;{{elementModel.label}}&nbsp;
+      </button>
+    </div>
+  `,
+  styles: [
+    '::ng-deep app-spell-correct .small-input div.mat-form-field-infix {border-top: none; padding: 0.75em 0 0.25em 0;}'
+  ]
+})
+export class SpellCorrectComponent extends FormElementComponent {
+  elementModel!: SpellCorrectElement;
+  @ViewChild(MatInput) inputElement!: MatInput;
+
+  strikethrough(): boolean {
+    if (this.inputElement) {
```

Comment:
Auch hier wieder unklar, warum das if benötigt wird.
Außerdem scheint mir kann man die Funktion massiv verkürzen. Null und Undefined kannst du auf jeden Fall in einem Schritt machen. Ich habe es nicht ausprobiert, aber mir scheint hier würde "return value.length > 0;" reichen. Der Rest ist überflüssig.